app/src/ui/changes/changes-list.tsx

```diff
@@ -612,6 +612,12 @@ export class ChangesList extends React.Component<
     const prepopulateCommitSummary =
       filesSelected.length === 1 && !repository.isTutorialRepository
 
+    const hasWritePermissionForRepository =
+      this.props.repository.gitHubRepository &&
+      this.props.repository.gitHubRepository.permissions === 'read'
+        ? false
+        : true
```

Comment:
yup, that's desired! `permissions` is `null` when we don't know what the permissions are, so i'm defaulting to the assumption that they do have write access, similar to your branch protection logic. 😸

Reply:
did a little cleanup in https://github.com/desktop/desktop/pull/8668/commits/e70e80123e82a8614b5cbcf5b68a4e2cae43afc6

and caught a bug where if the github repo was `null`, it would say the user didn't have push access!